docs/Migrations.rst

```diff
@@ -0,0 +1,260 @@
+Migrations
+==================
+
+Neo4j.rb provides an ``ActiveRecord``-like migration framework and a set of helper methods to manipulate both database schema and data.
```

Comment:
For people that are new to Neo4j they may need some context since the use of our migrations isn't quite like `ActiveRecord`'s.  Maybe saying something like "Neo4j does not have a set schema like relational databases, but sometimes changes to the schema and the data are required.  To help with this, Neo4j.rb provides an ``ActiveRecord``-like" etc...